src/prefect/orion/api/flow_runs.py

```diff
@@ -260,3 +260,43 @@ async def set_flow_run_state(
         response.status_code = status.HTTP_201_CREATED
 
     return orchestration_result
+
+
+@router.post("/{id}/restart")
```

Comment:
If we think making this an orchestration rule is correct, I don't see the benefit of this route? Seems like setting the flow run state to `AwaitingRestart` is all that's need?

Reply:
the new route is mostly a convenient way to specify an entirely different orchestration policy: for instance we'd be moving the state of a run "backwards" which is normally prohibited by the standard policy

Reply:
This "feels" very wrong to me -- either restarting flows is an expected orchestration path, or it's an exception, but needing to load an entire new policy to handle it goes against the design of the rules engine.

For example, the `PreventRestartingFlowsWithoutDeployments` rule is defined on every single transition, meaning it can not be used except with the `FlowRestartPolicy` that is loaded only on specific transitions called from a specific route -- the purpose of the rules engine is to provide a generic policy that embraces all expected transitions. In this model, `PreventRestartingFlowsWithoutDeployments` should be defined on `[<acceptable prior states>] → [AwaitingRetry]` only and included in the global flow run policy. It would only be enforced when attempting to retry.

If the rule about moving the run state backwards is too strict to account for this new behavior, then that rule can be relaxed rather than introducing an entirely new orchestration state machine to work around it.